src/WebJobs.Script/Description/Workers/WorkerFunctionDescriptorProvider.cs

```diff
@@ -20,13 +20,15 @@ internal abstract class WorkerFunctionDescriptorProvider : FunctionDescriptorPro
     {
         private readonly ILoggerFactory _loggerFactory;
         private IFunctionInvocationDispatcher _dispatcher;
+        private IScriptJobHostEnvironment _scriptJobHostEnvironment;
 
         public WorkerFunctionDescriptorProvider(ScriptHost host, ScriptJobHostOptions config, ICollection<IScriptBindingProvider> bindingProviders,
-            IFunctionInvocationDispatcher dispatcher, ILoggerFactory loggerFactory)
+            IFunctionInvocationDispatcher dispatcher, ILoggerFactory loggerFactory, IScriptJobHostEnvironment scriptJobHostEnvironment)
```

Comment:
>IScriptJobHostEnvironment [](start = 84, length = 25)

Usage of this to trigger shutdown is being reevaluated as part of a different PR and this might change accordingly